lib/dl_zitadel/dl_zitadel/middlewares/aiohttp.py

```diff
@@ -21,6 +23,112 @@
 class AioHTTPMiddleware:
     _client: clients.ZitadelAsyncClient
     _token_storage: services.ZitadelAsyncTokenStorage
+    _get_token_from_headers: bool = attr.ib(default=True)
+
+    async def process_tokens_from_cookies(
+        self,
+        app_request: dl_api_commons_aiohttp_aiohttp_wrappers.DLRequestBase,
+    ) -> aiohttp_web.StreamResponse | None:
+        req = app_request.request
+        zitadel_cookie_str = req.cookies.get(middlewares_models.ZitadelCookies.ZITADEL_COOKIE)
+        if zitadel_cookie_str is None:
+            LOGGER.info("Zitadel cookie is missing")
+            raise aiohttp_web.HTTPUnauthorized()
+        zitadel_cookie = json.loads(base64.b64decode(zitadel_cookie_str))
+
+        access_token, refresh_token = (
+            zitadel_cookie["passport"]["user"]["accessToken"],
+            zitadel_cookie["passport"]["user"]["refreshToken"],
+        )
+
+        if access_token is None:
+            LOGGER.info("Access token is missing")
+            raise aiohttp_web.HTTPUnauthorized()
+
+        token_introspect_result = await self._client.introspect(token=access_token)
+
+        if not token_introspect_result.active:
+            LOGGER.info("Access token is not active")
+
+            if refresh_token is None:
+                LOGGER.info("Refresh token is missing")
+                raise aiohttp_web.HTTPUnauthorized()
+
+            token_introspect_result = await self._client.introspect(token=refresh_token)
+            if not token_introspect_result.active:
+                LOGGER.info("Refresh token is not active")
+                raise aiohttp_web.HTTPUnauthorized()
+
+            refreshed_tokens = await self._client.refresh_token(refresh_token)
+            LOGGER.info("Tokens successfully refreshed")
+            access_token, refresh_token = (
+                refreshed_tokens.access_token,
+                refreshed_tokens.refresh_token,
+            )
+        else:
+            LOGGER.info("Access token is active")
+
+        zitadel_cookie["passport"]["user"]["accessToken"] = access_token
+        zitadel_cookie["passport"]["user"]["refreshToken"] = refresh_token
+
+        zitadel_cookie = base64.b64encode(json.dumps(zitadel_cookie).encode("ascii")).decode("ascii")
+        app_request.replace_temp_rci(
+            attr.evolve(
+                app_request.temp_rci,
+                user_id=token_introspect_result.sub,
+                user_name=token_introspect_result.username,
+                tenant=dl_api_commons_base_models.TenantCommon(),
+                auth_data=middlewares_models.ZitadelAuthData(
```

Comment:
better go further with the same method as in b2b case, using service_access_token and user_access_token